specs/searchable-asn-group/searchable-asn-group.md

```diff
@@ -0,0 +1,156 @@
+# Searchable ASN Filters (ff-asn-group)
+
+Some data sets have huge amounts of filters per filter group (150+). Too many to display all at once and too many to browse by eye.
+
+To improve usability the `ff-asn-group` gets an _optional_ search field. It is scoped to its associated filter group.
+
+
+## General
+
+This feature is an addition to the existing `ff-asn-group`. HTML examples are limited to new parts and minimal context for focus.
+
+The new search field is always the first element inside the collapsible `.ffw-wrapper` container. This means the filter group must be expanded in order to use it.
+
+Users may specify a custom search field template. It is defined via the `slot="filterSearch"` attribute. As it is a slot, its location inside the `ff-asn-group` template is irrelevant.
+
+## Features
+
+#### search as you type
+
+Filter items are searched for while user is typing in search field. While search term is present, filter group is kind of in _"search mode"_ compared to _"regular mode"_.
+
+- matching with `new RegExp(searchTerm, "gi")` or better algorithm
+- matched substrings are wrapped in `<span class="ffw-query">`
+- while search term is present, _"show more"_ disappears, matches of detailed and hidden links are merged into one list
+- _match-head_ results are listed first, then all other matches
+  - e.g.: search term: `bc`, result: `[bcd, bce, abc, cbc]`
+  - order of items within the _match-head_ and _match-any_ groups remains unchanged from order in FF response
+- not matching items are hidden
+- no special treatment for already selected filters
+  - selected filters are filtered just like all others
+
+If a filter item (selected or unselected) is clicked while in _"search mode"_, the search field is cleared and the component returns to _"regular mode"_.
+
+Any form of clearing the search field sets the component back to _"regular mode"_. This could, for example, be pressing the "X" on a `<input type="search">`.
+
+
+## Default HTML
+
+The search field is only rendered if the attribute `searchable-from` is set with a valid number on `ff-asn`.
+
+If `searchable-from` is not set or has an invalid value, the search field will not be rendered.
+
+#### Setup
+```html
+<ff-asn searchable-from="31">
+</ff-asn>
+```
+
+#### Rendered
+
+Assuming there are `31` or more links (_detailed_ and _hidden_ combined) in this group.
+
+```html
+<ff-asn searchable-from="31">
+    <ff-asn-group>
+        <div class="ffw-asn-group-container">
+            <div class="ffw-wrapper">
+
+                <div slot="filterSearch"><input></div>  <!-- this is the new part -->
```

Comment:
**Just an idea:** Maybe a developer needs to display something like `"45 of 280 brands matches your search"`. Might be a useful information for the end user if the list is quite long. This can be a good hint to add one ore more characters to the search to narrow the list further down. If it is easy to implement you could provide these numbers to be used within mustache.

```html
<div slot="filterSearch">
  <input type="search">
  <span>You see {{numberOfFilteredOptions}} of {{totalNumberOfFilterOptions}} brands</span>
</div>
```

Reply:
We'll try to think of a way to implement this.

By the way, `totalNumberOfFilterOptions` can be done already now.
You would subscribe to the [ResultDispatcher's](https://web-components.fact-finder.de/api/3.x/core-result-dispatcher) `asn` topic and then add the property to the group elements by summing `group.elements.length` and `group.selectedElements.length`.

Reply:
After spending some time thinking about how to implement this feature I feel this needs more consideration. Therefore, I would like to skip it this time and engage it in the specs' second iteration.

Reply:
As this is only based on a short idea, it's not critical for me and I'm happy if it will be discussed again any time in the future. This is not even a requirement in our current project, so some UX experts should evaluate the value of such a feature.